downloadservice/app.py

```diff
@@ -353,3 +365,77 @@ def oauth_callback():
     next_url = auth.get_next_url(cookie_state) or url_prefix
     response = make_response(redirect(next_url))
     return response
+
+
+webdav_methods = ['GET', 'HEAD',  'MKCOL', 'OPTIONS',
+                  'PROPFIND', 'PROPPATCH', 'PUT', 'TRACE',
+                  # 'LOCK', 'UNLOCK', 'POST', 'DELETE', 'MOVE',
+                  ]
+
+
+@app.route(url_prefix + "/webdav", defaults={'path': ''},
+           methods=webdav_methods)
+@app.route(url_prefix + "/webdav/<path:path>", methods=webdav_methods)
+@authenticated
+def webdav(user, path):
+    API_HOST = urljoin_multipart(
+        app.config['CTADS_UPSTREAM_ENDPOINT'],
+        app.config['CTADS_UPSTREAM_BASEPATH'],
+    )
+
+    def is_chunked():
+        return request.headers.get('transfer-encoding', '').lower() \
+            == 'chunked'
+
+    def request_datastream():
+        while (buf := request.stream.read(default_chunk_size)) != b'':
+            yield buf
+
+    if request.method in ['PUT', 'MKCOL'] and not path.startswith(
```

Comment:
what will happen here with path with ".."?

Reply:
also, what about `PROPPATCH`, it's also destructive

Reply:
It is resolved by flask and and correctly handled, I added a test